src/libraries/System.Text.Json/tests/TrimmingTests/EnumConverterTest.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SerializerTrimmingTest
+{
+    /// <summary>
+    /// Tests the serializer (de)serializes enums appropriately,
+    /// and that the enum converter factory is linker-safe.
+    /// </summary>
+    internal class Program
+    {
+        static int Main(string[] args)
+        {
+            string json = JsonSerializer.Serialize(new ClassWithDay());
+            return json == @"{""Day"":5}" ? 100 : -1;
+        }
+    }
+
+    internal class ClassWithDay
+    {
+        public DayOfWeek Day { get; set; } = DayOfWeek.Friday;
```

Comment:
Might be better not to touch the properties and leave them at their default value (this applies to all the other tests too) - linker can potentially have heuristics that say: if a setter is kept, also keep the getter.

Either keep the properties at their default values, or instead of auto properties, implement this with an explicit private backing field and only ever touch the backing field with C# - don't reference the properties from C# at all.

Reply:
(Basically, we need to make the properties as unreferenced as possible so that the linker is as free to strip it and the only reason why it didn't strip it are the annotations.)